docs/gguf.md

```diff
@@ -74,12 +74,18 @@ enum gguf_metadata_value_type: uint32_t {
     ///
     // Arrays can be nested, and the length of the array is the number of elements in the array, not the number of bytes.
     GGUF_METADATA_VALUE_TYPE_ARRAY = 9,
+    // The value is a 64-bit unsigned little-endian integer.
+    GGUF_METADATA_VALUE_TYPE_UINT64 = 10,
+    // The value is a 64-bit signed little-endian integer.
+    GGUF_METADATA_VALUE_TYPE_INT64 = 11,
+    // The value is a 64-bit IEEE754 floating point number.
+    GGUF_METADATA_VALUE_TYPE_FLOAT64 = 12,
 }
 
 // A string in GGUF.
 struct gguf_string_t {
     // The length of the string, in bytes.
-    uint32_t len;
+    uint64_t len;
```

Comment:
Eh, I'm not sure about this change. I don't have the version info when parsing strings. It will become very ugly to backward comp this

Reply:
`uint32_t` should be enough for string lengths

Reply:
Yes, I wasn't sure about that either. The two reasons I did it are a) for consistency with arrays and b) it is *theoretically* possible that someone might embed more than 4 gigabytes of string data in a metadata value. 

I can undo this, but how hard would it be to switch string parsing based on version?

Reply:
Nvm, let me try something

Reply:
I prefer to leave it `uint32_t` and will gladly revert the change I just made.
If we can't think of a meaningful application, lets leave it `uint32_t`. The user can always split their data into 4G chunks

Reply:
This is a bit fat fetched, but maybe in the future we could store datasets in GGUF format, and then maybe strings larger than 4 GiB could make sense.

Reply:
strings are uint64_t in impl PR now.

Reply:
Since we have 64-bit arrays, I think we can easily keep 32-bit long strings

Reply:
Let's have it 64-bit for consistency